src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -705,7 +759,13 @@ codet java_bytecode_convert_methodt::convert_instructions(
             statement=="invokevirtual" ||
             statement=="invokestatic")
     {
-      const bool use_this(statement!="invokestatic");
+      // Remember that this is triggered by an assertion
+      if(statement=="invokespecial" &&
+        as_string(arg0.get(ID_identifier)).find("AssertionError")!=std::string::npos)
```

Comment:
is `id2string` now
this seems to be how `assert` is compiled in Java
```
      35: getstatic     #11                 // Field $assertionsDisabled:Z
      38: ifne          53
      41: iload_1       
      42: ifeq          53
      45: new           #12                 // class java/lang/AssertionError
      48: dup           
      49: invokespecial #13                 // Method java/lang/AssertionError."<init>":()V
      52: athrow        
```
the `invokespecial` bytecode is used to initialize an object instance, so while this is a heuristic, it seems to be quite specific to detect the case we are looking for